### PR TITLE
Ruby 2.2 / Rack 2 and maintenance update

### PR DESCRIPTION
**Pull request by pond**
Includes `bundle exec rake rerdoc` update spam to track new version number.

Note the `Gemfile` split and the extraction of the _development mode only_ dependency on Alchemy Flux in the `.gemspec` file. This now moves into `Gemfile` and `rack_1.gemfile`, wherein specific versions are requested. The main `Gemfile[.lock]` works with Ruby 2.2 upwards via Rack 2 and Flux 1.1, with `.ruby-version` ensuring RBEnv-based Hoodoo developers get this version by default. For Travis only, `rack_1.gemfile` locks itself to older versions of Flux allowing Rack 1.6 to be used and thus Ruby 2.1.9 to be run for the test suite.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-  - 2.2.5
-  - 2.3.1
+matrix:
+  include:
+    - rvm: 2.3.1
+      gemfile: Gemfile 
+    - rvm: 2.2.5
+      gemfile: Gemfile
+    - rvm: 2.1.9
+      gemfile: rack_1.gemfile
```
Comment (mbdietrich) [inline .travis.yml]: Can you explain this change a little more please?

Comment (pond) [inline .travis.yml]: See the full description for the PR. Copypasta:

"Note the Gemfile split and the extraction of the development mode only dependency on Alchemy Flux in the .gemspec file. This now moves into Gemfile and rack_1.gemfile, wherein specific versions are requested. The main Gemfile[.lock] works with Ruby 2.2 upwards via Rack 2 and Flux 1.1, with .ruby-version ensuring RBEnv-based Hoodoo developers get this version by default. For Travis only, rack_1.gemfile locks itself to older versions of Flux allowing Rack 1.6 to be used and thus Ruby 2.1.9 to be run for the test suite."

The bottom line is that Rack 2 only works on Ruby 2.2 upwards, but some Hoodoo users might be on the Ruby 2.1 series and content with Rack 1.6, and/or face difficulties in upgrading Ruby.

Comment (mbdietrich) [inline .travis.yml]: Okay, I'm happy now that I've had compared it with the original Gemspec. +1.
